### DISCUSSION TITLE
[HELP] MongoEngine and FastAPI Integration : GET API maximum recursion depth exceeded/

### DISCUSSION THREAD

**Discussion by naheedroomy** [fastapi/fastapi#4005] [Questions] [answered]
Getting this error when running a simple GET Api on a Mongo Collection. the collection has around 20 records right now.

`RecursionError: maximum recursion depth exceeded in comparison
`

My code for this api is included below. 

from pydantic import BaseModel
```
from mongoengine import *
from fastapi import FastAPI
from fastapi.encoders import jsonable_encoder
from fastapi.responses import JSONResponse

app = FastAPI()


connect('pythonmongodb')


class User(BaseModel):
    username: str
    email: str
    password: str


class users(Document):
    username = StringField(required=True)
    password = StringField(required=True, min_length=4)
    email = EmailField(required=True, unique=True)


@app.get('/users', response_model=list[User], tags=["users"])
async def find_all_users():
    abc = jsonable_encoder(users.objects)
    return JSONResponse(content=abc)
```

**Comment by negadive** [comment, marked as answer]
`RecursionError` caused by applying `jsonable_encoder` on your `users.objects` (instance of `<class 'mongoengine.queryset.queryset.QuerySet'>`), it made dict by applying `dict()` or `var()` to`users.objects` and iterate unnecessary keys that could contains infinite loop

you would like to use orm mode to your response schema
```python
class User(BaseModel):
    username: str
    email: str
    password: str

    class Config:
        orm_mode = True
```
[pydantic docs about orm_mode](https://pydantic-docs.helpmanual.io/usage/models/#orm-mode-aka-arbitrary-class-instances)
[fastapi docs about orm_mode](https://fastapi.tiangolo.com/tutorial/sql-databases/#use-pydantics-orm_mode)

and `find_all_users` would be just like this
```python
async def find_all_users():
    return list(users.objects)
```

this will making dicts by `GetterDict` instead of `jsonable_encoder`
